core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java

```diff
@@ -106,6 +113,8 @@ public Builder addAllConfig(Map<String, String> properties) {
 
     public LoadTableResponse build() {
       Preconditions.checkNotNull(metadata, "Invalid metadata: null");
+      // This is causing errors in TestRESTCatalog via CatalogHandlers
+      // Preconditions.checkNotNull(metadataLocation, "Invalid metadata location: null");
```

Comment:
There's a call to the builder somewhere that's used during the `CatalogTests` that is creating a `LoadTableResponse` without a metadata location.

I think it's somewhere related to here https://github.com/apache/iceberg/blob/e0ff3c6e9876ce04c890908e375c876d14240c3e/core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java#L160-L195, or some other create transaction that doesn't commit right away. Will investigate further.

Reply:
It's this `LoadTableResponse` https://github.com/apache/iceberg/blob/e0ff3c6e9876ce04c890908e375c876d14240c3e/core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java#L483-L484, which is called in the failing tests via `catalog.buildTable(TABLE, OTHER_SCHEMA).createTransaction()` which then fails the builder because there's no metadata location yet.

Reply:
We've determined that `metadata-location` is not required (can be null), due to this very case.